bazel/astore/defs.bzl

```diff
@@ -114,28 +114,40 @@ TODO(jonathan): add support for the "dir" attribute.
 )
 
 def _astore_download(ctx):
-    output = ctx.actions.declare_file(ctx.attr.download_src.split("/")[-1])
-    command = ("%s download --no-progress -o %s %s" %
-               (ctx.executable._astore_client.path, output.path, ctx.attr.download_src))
+    if ctx.attr.output:
+      output = ctx.outputs.output
+    else:
+      output = ctx.actions.declare_file(ctx.attr.download_src.split("/")[-1])
+    command = ("%s download --no-progress --overwrite -o %s" %
+               (ctx.executable._astore_client.path, output.path))
+    execution_requirements = {
+            # We can't run these remotely since remote workers won't have
+            # credentials to fetch from astore.
+            "no-remote": "Don't run remotely or cache remotely",
+            "requires-network": "Downloads from astore",
+            "timeout": "%d" % ctx.attr.timeout,
+        }
     if ctx.attr.arch:
         command += " -a " + ctx.attr.arch
+    if ctx.attr.uid:
+        command += " --force-uid %s" % ctx.attr.uid
+    else:
+        command += " %s" % ctx.attr.download_src
+        execution_requirements["no-cache"] = "Not hermetic, since uid was not specified."
```

Comment:
I'd also move the `no-remote` here? eg, if the artifact is cachable, it's ok if it's remote as well?